scormxblock/scormxblock.py

```diff
@@ -196,50 +285,67 @@ def studio_submit(self, request, suffix=''):
 
         return Response(json.dumps({'result': 'success'}), content_type='application/json')
 
-    @XBlock.json_handler
-    def scorm_get_value(self, data, suffix=''):
-        name = data.get('name')
-        if name in ['cmi.core.lesson_status', 'cmi.completion_status']:
-            return {'value': self.lesson_status}
-        elif name == 'cmi.success_status':
-            return {'value': self.success_status}
-        elif name == 'cmi.core.lesson_location':
-            return {'value': self.lesson_location}
-        elif name == 'cmi.suspend_data':
-            return {'value': self.suspend_data}
-        else:
-            return {'value': self.data_scorm.get(name, '')}
+    @XBlock.handler
+    def scorm_get_values(self, request=None, suffix=None):
+        values = {
+            'cmi.core.lesson_status': self.lesson_status,
+            'cmi.completion_status': self.lesson_status,
+            'cmi.success_status': self.success_status,
+            'cmi.core.lesson_location': self.lesson_location,
+            'cmi.suspend_data': self.suspend_data,
+        }
+
+        data = {}
+
+        for key, value in self.data_scorm.iteritems():
+            if key in READ_MODEL_DATA:
+                data[key] = value
+            elif key.startswith('cmi.interactions.') and validate_property(key, READ_MODEL_DATA):
+                data[key] = value
+            elif key.startswith('cmi.objectives.') and validate_property(key, READ_MODEL_DATA):
+                data[key] = value
+
+        values.update(data)
+        return Response(json.dumps(values), content_type='application/json')
 
     @XBlock.json_handler
-    def scorm_set_value(self, data, suffix=''):
+    def scorm_set_values(self, data, suffix=''):
         context = {'result': 'success'}
-        name = data.get('name')
 
-        if name in ['cmi.core.lesson_status', 'cmi.completion_status']:
-            self.lesson_status = data.get('value')
-            if self.has_score and data.get('value') in ['completed', 'failed', 'passed']:
-                self.publish_grade()
+        for name, value in data.iteritems():
+            if name in ['cmi.core.lesson_status', 'cmi.completion_status']:
+                self.lesson_status = value
+                if self.has_score and value in ['completed', 'failed', 'passed']:
+                    self.publish_grade()
+                    context.update({"lesson_score": self.lesson_score})
+
+            elif name == 'cmi.success_status':
```

Comment:
This could be written as: elif name == 'cmi.success_status' and self.has_score: do this, since you are not doing anything else if not self.has_score. @andrey-canon